app/routes/task.py

```diff
@@ -1,44 +1,53 @@
 from flask import Blueprint, request, jsonify
 from app.services import *
+from app.decorators import *
 
 task_bp = Blueprint('add_task', __name__)
 
 
 @task_bp.route('/', methods=['POST'])
-def add_task():
+
+@token_required
+@requires_role_manager()
+def add_task(current_user):
     data=request.get_json()
     if not data.get('title') or not data.get('description'):
         return jsonify({'error': 'Title and description cannot be empty'}), 400
-    result = add(title=data.get('title'), description=data.get('description'))
+    result=add(data)
     return jsonify(result),201
 
 
 @task_bp.route('/<task_id>', methods=['GET'])
-def view_task(task_id):
+@token_required
+def view_task(current_user, task_id):
     if task_id is None:
         return jsonify({'error': 'Task ID is required'}), 400
-    task = view(task_id)
-    if task:
-        return jsonify(task)
-    else:
-        return jsonify({'error': 'Task not found'}), 404
-    
+    if current_user.role == Role.admin or current_user.role == Role.manager:
+        task = view(task_id)
+        if task:
+            return jsonify(task)
+        else:
+            return jsonify({'error': 'Task not found'}), 404
+        
     
 @task_bp.route('/', methods=['GET'])
-def view_all_tasks():
+@token_required
+def view_all_tasks(current_user):
     tasks = viewall()
     return jsonify(tasks),201
 
 @task_bp.route('/<int:task_id>', methods=['DELETE'])
-def delete_task(task_id=None):
+@token_required
+def delete_task(current_user,task_id=None):
     if task_id is None:
         return jsonify({'error': 'Task ID is required'}), 400
     result = delete(task_id)
     return jsonify(result),201
 
 
 @task_bp.route('/<int:task_id>', methods=['PUT'])
-def update_task(task_id):
+@token_required
+def update_task(current_user,task_id):
```

Comment:
Remove this

if task_id is None:
        return jsonify({'error': 'Task ID is required'}), 400
        
Only check for it task exists or not